problem1.java

```diff
@@ -0,0 +1,16 @@
+class arrayPartition {
+    public int arrayPairSum(int[] nums) {
+        if(nums == null || nums.length == 0) return 0;
+        
+        Arrays.sort(nums);
+        int sum = 0;
+        
+        for(int i = 0; i < nums.length; i = i + 2){
+            sum = sum + nums[i];
+        }
+        return sum;
+    }
+}
+
+//time complexity O(nlog n)
+//space complexity O(1)
```

Comment:
The mentioned space complexity is wrong. Please check it once again.

Reply:
I don't use any extra space, whatever given I just sorted and use only one variable 'sum' so not sure how space complexity would change?

Reply:
You are using Arrays.sort() for sorting which takes up log N space